543.py

```diff
@@ -0,0 +1,31 @@
+# Definition for a binary tree node.
+# class TreeNode:
+#     def __init__(self, val=0, left=None, right=None):
+#         self.val = val
+#         self.left = left
+#         self.right = right
+class Solution:
+    def diameterOfBinaryTree(self, root: TreeNode) -> int:
+        def getMaxDepth(node: TreeNode) -> int:
+            if not node:
+                return 0
+
+            if not node.left and not node.right:
+                return 1
+
+            return max(getMaxDepth(node.left), getMaxDepth(node.right)) + 1
+
+        stack = [root]
+        diameter = 0
+
+        while stack:
```

Comment:
Did you notice this structure (from line 18 to line 29) is very similar to the BFS template?

Reply:
This is pre-order traversal, right?

Reply:
It looks like it's post-order--you get your results from both children first and then update `diameter` of the current node.

Reply:
This `while stack:` structure is an iterative DFS template.